plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -246,6 +249,23 @@ protected void refreshFromMetadataLocation(String newLocation)
         shouldRefresh = false;
     }
 
+    private boolean shouldBeAborted(Throwable failure)
+    {
+        if (failure instanceof org.apache.iceberg.exceptions.NotFoundException) {
+            return true;
+        }
+        if (failure.getCause() != null) {
```

Comment:
> With Glue catalog, the metadata file is read using the TrinoS3FileSystem....

`org.apache.iceberg.hadoop.HadoopInputFile#newStream` already throws `NotFoundException`


I don't follow the reasoning behind the low-level check of the cause of the exception.

Reply:
As discussed offline, we will be checking for the `FileNotFoundException` exception here for `abortOn` instead of converting java's `FileNotFoundException` to Iceberg's `NotFoundException` in `ForwardingSeekableInputStream`.

For other abortable exceptions, I have created an issue https://github.com/trinodb/trino/issues/16908